AcademyChallenge/Presentation/Scenes/AvatarViewController.swift

```diff
@@ -7,147 +7,80 @@
 
 import UIKit
 
-class AvatarViewController: UIViewController {
-	
-	var avatarService: LiveAvatarStorage?
-	var avatarList: [Avatar] = []
-	
-	
-	lazy var collectionView: UICollectionView = {
-			let s = UICollectionView(frame: .zero, collectionViewLayout: UICollectionViewFlowLayout())
-			return s
-		}()
+public protocol AvatarViewControlerDelegate: AnyObject {
+    func navigateToFirstPage()
+}
 
+class AvatarViewController: BaseGenericViewController<AvatarView> {
+    public weak var delegate: AvatarViewControlerDelegate?
```

Comment:
why is the `public` modifier needed here?

Reply:
I don't need actually